### ISSUE TITLE
Font scaling issues with two monitors on Windows with different Scale setting.

### ISSUE THREAD

**Issue by bilderbuchi** [OpenModelica/OpenModelica#12523]
Is there maybe some weird scaling/dpi issue going on (see also #12519)? 
I'm recently getting some very weird font sizes on some elements, on a 1920x1080 secondary monitor (Windows scaling setting 100%) connected to a laptop with a smaller 2560x1600 screen (Windows scaling setting 175% makes everything legible across monitors)
![image](https://github.com/OpenModelica/OpenModelica/assets/327442/7fb31d46-e4db-4076-ae0f-aa380effe8f6)

This is what it looks like when starting OM with **both** screens active (my default situation) -- some UI text across OMEdit is illegibly small:
![image](https://github.com/OpenModelica/OpenModelica/assets/327442/97c2f2cc-251f-4cb9-915a-9d612a4ef853)

This is when starting with only the secondary active, those small texts look well-sized:
![image](https://github.com/OpenModelica/OpenModelica/assets/327442/96cc41e8-d7bc-42f0-8d1d-d0b0d98423b6)

This is with only the small high-res laptop screen active -- smaller texts, a bit uncomfortable but still legible because it's very crisply rendered:
![image](https://github.com/OpenModelica/OpenModelica/assets/327442/f5c2b4c1-5cf7-453e-bb27-37303ebe6f9a)

Starting in single-screen, then extending degrades the situation back to the first screenshot.

I suspect some configuration in OMEdit/Qt started to interact badly with the scale settings in the [Windows scale and layout setting](https://www.windowscentral.com/how-set-custom-display-scaling-setting-windows-10#change_display_scaling_default_settings_windows10) (like some other software does, too).

**Comment by bilderbuchi** [comment]
> Starting in single-screen, then extending degrades the situation back to the first screenshot.

I just realised that only when starting with only the laptop screen (scale 175%) active, then extending, goes back to the bad first situation.
If I start with only the _external (scale 100%) monitor_ active, then extend to both screens, I get the (good) situation 2. So that is to me a reasonable workaround until this can be fixed.

**Comment by casella** [comment]
@adeas31 as time passes, screen resolutions will go from 4K to 8K to 16K and even more extreme values, so we need a sustainable path to support them. Do you think you can manage?

**Comment by casella** [comment]
See also #13470

**Comment by bilderbuchi** [comment]
A quick test using the experimental build made available by @adrpo https://github.com/OpenModelica/OpenModelica/issues/13470#issuecomment-2604457007 shows that this issue seems to be resolved with that PR.

**Comment by casella** [comment]
Good to hear!

I understand we need to wait for a good enough html browser to become available in Qt6 before we take the plunge, or do I miss something?

**Comment by adeas31** [comment]
> I understand we need to wait for a good enough html browser to become available in Qt6 before we take the plunge, or do I miss something?

Yeah that's true. The idea behind PR #13478 is just to show that the scaling issues we are having are related to Qt and there is nothing more we can do about it. Our code is already compatible with Qt6 and we are regularly compiling and testing against Qt6. Qt6 webengine is not yet available for Windows MSYS. And as far as I remember, the one available for Linux ubuntu crashes. So once its available we can switch to Qt6 but until then we have to live with these scaling issues.

**Comment by casella** [comment]
OK. I checked the [Qt support timeline](https://endoflife.date/qt):

![Image](https://github.com/user-attachments/assets/d348887c-9e1f-45e6-a56d-eb8e6d1a341e)

and it seems that Qt5 is still on life support (pun intended) for a few more months, but it's really approaching the end.

This [post on stackoverflow](https://stackoverflow.com/questions/67646627/why-is-there-no-qtwebengine-in-the-qt6) said in 2021 that webengine support would come with Qt 6.2. [This post](https://www.qt.io/blog/qt-webengine-in-qt-6) says WebEngine has been out since Qt 6.2 in 2022. From the [QT documentation](https://doc.qt.io/qt-6/qtwebengine-index.html) it seems to be firmly available in Qt 6.8. I see Qt 6.8.1-1 in [mingw-w64-x86_64-qt6-webview](https://packages.msys2.org/packages/mingw-w64-x86_64-qt6-webview). I didn't find any recent posts complaining about Qt6 webengine issues.

Do I miss something?

**Comment by casella** [comment]
OK.

> And as far as I remember, the one available for Linux ubuntu crashes.

This shouldn't be the case, however, right?

**Comment by adeas31** [comment]
Yes that is the case when I tested it last time. See https://github.com/OpenModelica/OpenModelica/pull/11581#issuecomment-2059717470

Maybe a newer Qt version is available for Ubuntu now that works. Someone needs to test it.

**Comment by casella** [comment]
I understand you can do apt upgrade of the installed qt, can't you?

**Comment by casella** [comment]
@adrpo will check if there are some recent fixes to Qt5 that solve this problem. He can also try to build the Web engine himself, doing it on GitHub currently takes too much time.
